doc/feature/spikeReport.md

```diff
@@ -0,0 +1,327 @@
+important characteristic of the current proposal is that all operations can only
+move forward, so each operation starts where the previous one left over.
+This way there is no need to make any distinction between the file and stream
+based cases. Random access is partially supported by a function that does
+forward skip.
+
+The current proposal defers important design decisions to the higher level API,
+but the its benefit is that it provides a relatively simply API for the most
+critical part.
+
+## API
+
+Thread-safey is only guaranteed for const access.
+
+### Helper types
+
+    struct Spike
+    {
+        Spike( float timestamp_, uint32_t gid_ );
+        float timestamp;
+        uint32_t gid;
+    };
+
+    typedef std::vector< Spike > Spikes;
+
+### SpikeReport
+
+    SpikeReport : public boost::noncopyable
+    {
+    public:
+        SpikeReport( const URI& uri, int mode );
+
+        /*
+         * Open a report in read mode with a subset selection.
+         *
+         * @param uri
+         * @param subset The set of gids to be reported. This set should be
+         *        understood as a filter: any included GID which is not
+         *        actually part of the report will be silently ignored.
+         */
+        SpikeReport( const URI& uri, const GIDSet& subset );
+
+        /**
+         * Release all resources.
+         * Any threads waiting on futures will return immediately and calling
+         * get will throw std::runtime_error.
+         */
+        ~SpikeReport();
+
+        const URI& getURI();
+
+        /** ? */
+        float getStartTime() const;
+        float getEndTime() const;
+
+        /**
+         * @return the end time of the latest complete read/write operation or
+         *         UNDEFINED_TIMESTAMP if no operation has been issued.
+         *
+         * Read operations are deemed as complete when the returned future is
+         * ready.
+         *
+         * The time inverval to which getCurrentTime refers is open on the
+         * right. That means that upon completion of a read or write operation
+         * no spike read or written may have a timestamp >= getCurrentTime().
+         */
+        float getCurrentTime() const;
+
+        /**
+         * Read spikes until getCurrentTime becomes > min, the end of the report
+         * is reached or the report closed.
+         *
+         * This function fetches all the available data from the source.
+         * If no error occurs and the end is not reached, the min value passed
+         * is also guaranteed to be inside the time window of the data returned.
+         *
+         * Preconditions:
+         * - r.getState() is OK or ENDED.
+         * - The report was open in read mode.
+         * - There is no previous read or forward operation with a pending
+         *   future.
+         * - min >= getCurrentTime() or UNDEFINED_TIMESTAMP.
+         * Throw std::logic_error if one of the preconditions is not fulfilled.
```

Comment:
Yes, but it breaks the flow of text. I don't think the rendering is going to look correct if I add it.